tools/state-viewer/src/cli.rs

```diff
@@ -679,20 +717,79 @@ pub struct ViewTrieCmd {
     /// For format=pretty this measures depth in terms of key nibbles.
     #[clap(long)]
     max_depth: u32,
+    /// Limits how many entries are printed to the output.
+    #[clap(long)]
+    limit: Option<u32>,
+    /// Filters output to only show records of given type.
+    #[clap(long)]
+    rtype: Option<StateRecordType>,
+    /// Will only print nodes with key being a prefix of or lexicographically greater than `from`.
+    /// The argument should be wrapped in parentheses. Only works for `full` format.
+    #[clap(long)]
+    from: Option<String>,
+    /// Will only print nodes with key being lexicographically less than or equal to `to`.
+    /// The argument should be wrapped in parentheses. Only works for `full` format.
+    #[clap(long)]
+    to: Option<String>,
```

Comment:
note: escaped characters (e.g. `\x`) are parsed differently depending on whether the argument is provided in parentheses